llvm/include/llvm/ADT/Matrix.h

```diff
@@ -0,0 +1,339 @@
+//===- Matrix.h - Two-dimensional Container with View -----------*- C++ -*-===//
```

Comment:
Would it be possible to first have this code live somewhere local to `ConstraintSystem` and once it matures / gains more usage, promote it to ADT?

Reply:
Possible, since it's header-only. I would like some other opinions as well.

Reply:
I'd strongly prefer this option then. Also, please do not click 'resolve' on conversations that haven't been resolved.